CMakeLists.txt

```diff
@@ -47,11 +47,18 @@ option(TEST_INTEL_OPTIMIZATION "Test intel optimization and vectorization" OFF)
 option(TEST_ERROR_CHECKING "Enables the error checking unit test. This test will FAIL" OFF)
 option(CODE_COVERAGE "Enable code coverage reporting" OFF)
 
+# Default to an external Kokkos package if the submodule is not populated
+if(NOT EXISTS "${CMAKE_CURRENT_SOURCE_DIR}/external/Kokkos/CMakeLists.txt" AND NOT EXISTS "${Kokkos_ROOT}/CMakeLists.txt")
```

Comment:
@AndrewGaspar I'm a little confused by the choice of external does this seem right to you? In this case it looks like setting kokkos_root will default to non external, but I would have thought if you are setting kokkos root you are referring to kokkos package which is external to parthenon (not using the kokkos version packaged as a submodule within parthenon) regardless of whether it is a library or not.

The way this is implemented it looks like "PARTHENON_EXTERNAL_KOKKOS" Is being reserved for an external library only and even kokkos is installed somewhere outside of parthenon it is not considered external if it is pointing at the kokkos root directory.

Reply:
Maybe `PARTHENON_FIND_KOKKOS` instead?